### DISCUSSION TITLE
Question: How does Hasura deal correctly with the problem of opening too many connections to Amazon Aurora, MySQL and Postgresql DBs?

### DISCUSSION THREAD

**Discussion by patelatharva** [hasura/graphql-engine#9680] [Question]

**Comment by texfu** [comment]
Thx

**Comment by SamirTalwar** [comment]
Hey folks, just came across this question. I expect the OP has already got their answer but I would like to document it for any others that come across it.

Hasura has a configurable `max_connections` parameter (for PostgreSQL; other databases have similar configuration). You can set this for all PostgreSQL databases with the `HASURA_GRAPHQL_PG_CONNECTIONS` environment variable (see [Docs: Tune Hasura Performance](https://hasura.io/docs/latest/deployment/performance-tuning/) for details), or configure your database's `max_connections` setting using the Console, CLI, or API (see [Docs: Quickstart](https://hasura.io/docs/latest/databases/quickstart/)).

Please consult the appropriate docs for any other databases.

On Cloud, this connection pool is shared across Cloud nodes so that a global limit is enforced (softly; it may temporarily exceed the total by a few due to rebalancing). See [Docs: Elastic connection pools](https://hasura.io/docs/latest/databases/database-config/cloud-connection-pooling/) for more information on Cloud.

I'm going to close this discussion, but please feel free to re-open it if you need more clarification or have questions.
